src/js/game/hud/parts/mass_selector.js

```diff
@@ -148,14 +149,30 @@ export class HUDMassSelector extends BaseHUDPart {
             // copy code relies on entities still existing, so must copy before deleting.
             this.root.hud.signals.buildingsSelectedForCopy.dispatch(entityUids);
 
-            for (let i = 0; i < entityUids.length; ++i) {
-                const uid = entityUids[i];
-                const entity = this.root.entityMgr.findByUid(uid);
-                if (!this.root.logic.tryDeleteBuilding(entity)) {
-                    logger.error("Error in mass cut, could not remove building");
-                    this.selectedUids.delete(uid);
+            let canPaste = false;
+            const placer = this.root.hud.parts.blueprintPlacer;
+            if (placer) {
+                const blueprint = placer.currentBlueprint.get();
+                if (blueprint) {
+                    canPaste = blueprint.canAfford(this.root);
                 }
             }
+            if (canPaste) {
+                for (let i = 0; i < entityUids.length; ++i) {
+                    const uid = entityUids[i];
+                    const entity = this.root.entityMgr.findByUid(uid);
+                    if (!this.root.logic.tryDeleteBuilding(entity)) {
+                        logger.error("Error in mass cut, could not remove building");
+                        this.selectedUids.delete(uid);
+                    }
+                }
+            } else {
+                this.selectedUids = new Set();
+                this.root.hud.signals.notification.dispatch(
+                    T.ingame.notifications.massCutInsufficient,
+                    enumNotificationType.success
```

Comment:
Instead of using a notification here, could you show a dialog?

(`this.root.hud.parts.dialogs.showWarning`)

Reply:
I changed it to show confirm dialog instead of notification